api/views.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from habittracker.models import Habit, Result, User
+from .serializers import HabitSerializer, ResultSerializer, UserSerializer, HabitResultSerializer
+from rest_framework.generics import ListCreateAPIView, ListAPIView, RetrieveUpdateDestroyAPIView, RetrieveAPIView, CreateAPIView
+
+
+class HabitListView(ListCreateAPIView, RetrieveUpdateDestroyAPIView ):
+    #queryset:
+    queryset = Habit.objects.all()
+    #serializer:
+    serializer_class = HabitSerializer
+    
+    def perform_create(self, serializer):
+        serializer.save(app_user=self.request.user)
+
+class ResultListView(RetrieveUpdateDestroyAPIView, ListCreateAPIView):
```

Comment:
Because you are inheriting from the `RetrieveUpdateDestroyAPIView` this view is not working as a list view. You want to see a list of results filtered by those that match the habit pk in the url. If you take out that class and just use the ListCreateAPIView your queryset should work as intended.